jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/impl/query/preprocessor/UserAndGroupsTasksPreprocessorTest.java

```diff
@@ -138,6 +140,39 @@ public void testPotOwnerFilterNoIdentityProvider() {
                      dataSetLookup.getFirstFilterOp().getColumnFilterList().get(0).toString());
     }
 
+    @Test
+    public void testPotOwnerFilterComplexExpression() {
+        String userId = "userId";
+
+        when(userGroupCallback.getGroupsForUser(userId)).thenReturn(null);
+        when(identityProvider.getName()).thenReturn(userId);
+
+        String potOwner = "potOwner";
+        when(userGroupCallback.getGroupsForUser(potOwner)).thenReturn(Arrays.asList("role1", "role2"));
+
+        List<String> potOwners = new ArrayList<String>();
+        potOwners.add(potOwner);
+
+        QueryParam queryParam = new QueryParam(COL_ID, "IN", potOwners);
+        QueryParam queryParamTerm1 = new QueryParam("MY_COLUMN", "EQUALS_TO", Arrays.asList("1234"));
+        QueryParam queryParamTerm2 = new QueryParam("MY_COLUMN", "NOT_IN", Arrays.asList(1, 2, 3, 4));
+        QueryParam queryParamOR = new QueryParam(null, "OR", Arrays.asList(queryParamTerm1, queryParamTerm2, queryParam));
```

Comment:
So, in this test, the whole filter to preprocess is

(MY_COLUMN = 1234 OR MY_COLUMN not_in 1, 2, 3, 4 **OR** POTOWNER in potOwner)

but after preprocessing, the filters are split in a list(because more values have been added for POTOWNER column) but they are going to be rebuilt with AND, no?

((MY_COLUMN = 1234 OR MY_COLUMN not_in 1, 2, 3, 4) **AND** (POTOWNER in role1, role2, potOwner))

Or am I misunderstanding anything?

Reply:
correct. This should the right approach how this filter works.